source/server/connection_handler_impl.h

```diff
@@ -31,46 +34,33 @@ struct ListenerStats {
  * Server side connection handler. This is used both by workers as well as the
  * main thread for non-threaded listeners.
  */
-class ConnectionHandler final : NonCopyable {
+class ConnectionHandlerImpl : public ConnectionHandler, NonCopyable {
 public:
-  ConnectionHandler(Stats::Store& stats_store, spdlog::logger& logger, Api::ApiPtr&& api);
-  ~ConnectionHandler();
+  ConnectionHandlerImpl(Stats::Store& stats_store, spdlog::logger& logger, Api::ApiPtr&& api);
+  ~ConnectionHandlerImpl();
 
   Api::Api& api() { return *api_; }
   Event::Dispatcher& dispatcher() { return *dispatcher_; }
   uint64_t numConnections() { return num_connections_; }
```

Comment:
style nit:

We don't repeat comments for implementations (overrides), and put them at the end below other methods. We also put a small comment to organize the overrides. I will update the style guide with this info.

So it would go like:

class ConnectionHandlerImpl {
public:
  void somethingNotInInterface();

  // Network::ConnectionHandler
  void method() override;
}